### ISSUE TITLE
Handle exception when no shared ciphers are found.

### ISSUE THREAD

**Issue by johnnykv** [johnnykv/heralding#22]
I guess that this goes for all protocols wrapped in `s`.

```
Aug 18 06:53:09 heralding heralding[21523]: Traceback (most recent call last):
Aug 18 06:53:09 heralding heralding[21523]: File "/home/heralding/virtualenv/local/lib/python2.7/site-packages/gevent/greenlet.py", line 534, in run
Aug 18 06:53:09 heralding heralding[21523]: result = self._run(*self.args, **self.kwargs)
Aug 18 06:53:09 heralding heralding[21523]: File "/home/heralding/virtualenv/local/lib/python2.7/site-packages/gevent/baseserver.py", line 25, in _handle_and_close_when_done
Aug 18 06:53:09 heralding heralding[21523]: return handle(*args_tuple)
Aug 18 06:53:09 heralding heralding[21523]: File "/home/heralding/virtualenv/local/lib/python2.7/site-packages/gevent/server.py", line 126, in wrap_socket_and_handle
Aug 18 06:53:09 heralding heralding[21523]: ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
Aug 18 06:53:09 heralding heralding[21523]: File "/home/heralding/virtualenv/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 691, in wrap_socket
Aug 18 06:53:09 heralding heralding[21523]: ciphers=ciphers)
Aug 18 06:53:09 heralding heralding[21523]: File "/home/heralding/virtualenv/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 271, in __init__
Aug 18 06:53:09 heralding heralding[21523]: raise x
Aug 18 06:53:09 heralding heralding[21523]: SSLError: [SSL: NO_SHARED_CIPHER] no shared cipher (_ssl.c:590)
```

**Comment by johnnykv** [comment]
You should be able to reproduce it with sslclient. Use it to connect to a heralding protocol wrapped in ssl (https or pop3s).

**Comment by johnnykv** [comment]
Closed with 6ba2a9d
